docs/guides/visualization.rst

```diff
@@ -0,0 +1,287 @@
+
+.. code:: python3
+
+    import geopyspark as gps
+    import matplotlib.pyplot as plt
+    
+    from colortools import Color
+    from pyspark import SparkContext
+    
+    %matplotlib inline
+
+.. code:: python3
+
+    !curl -o /tmp/cropped.tif https://s3.amazonaws.com/geopyspark-test/example-files/cropped.tif
+
+.. code:: python3
+
+    conf = gps.geopyspark_conf(master="local[*]", appName="visualization")
+    pysc = SparkContext(conf=conf)
+
+.. code:: python3
+
+    raster_layer = gps.geotiff.get(layer_type=gps.LayerType.SPATIAL, uri="/tmp/cropped.tif")
+    tiled_layer = raster_layer.tile_to_layout(layout=gps.GlobalLayout(), target_crs=3857)
+
+
+.. _visualizing:
+
+Visualizing Data in GeoPySpark
+==============================
+
+Data is visualized in GeoPySpark by running a server which allows it to
+be viewed in an interactive way. Before putting the data on the server,
+however, it must first be formatted and colored. This guide seeks to go
+over the steps needed to create a visualization server in GeoPySpark.
+
+.. _pyramid:
+
+Pyramid
+-------
+
+The ``pyramid`` class represents a list of ``TiledRasterLayer``\ s that
+represent the same area where each layer is a level within the pyramid
+at a specific zoom level. Thus, as one moves up the pyramid (starting a
+level 0), the image will become more zoomed by a power of 2 for each
+level. It is this varying level of detail that allows an interactive
+tile server to be created from a ``Pyramid``. This class is needed in
+order to create visualizations of the contents within its layers.
+
+Creating a Pyramid
+~~~~~~~~~~~~~~~~~~
+
+There are currently two different ways to create a ``Pyramid`` instance:
+Through the ``TiledRasterLayer.pyramid`` method or by constructing it by
+passing in a ``[TiledRasterLayer]`` or
+``{zoom_level: TiledRasterLayer}`` to ``Pyramid``.
+
+Any ``TiledRasterLayer`` with a ``max_zoom`` can be pyramided. However,
+the resulting ``Pyramid`` may have limited functionality depending on
+the layout of the source ``TiledRasterLayer``. In order to be used for
+visualization, the ``Pyramid`` **must** have been created from
+``TiledRasterLayer`` that was tiled using a ``GlobalLayout`` and whose
+tile sizes are a power of 2.
```

Comment:
Again in effort to be more precise, tile sizes do not change from level to level in the pyramid, only the spatial resolution of those tiles. That is any given pixel of a tile covers larger and larger area as you go up the pyramid, decreasing in the resolution of the tile.